### PR TITLE
Upgrade to puppeteer 9.1.1

### LINKED ISSUE #6876: Update `puppeteer` to latest version
## Task Description

Need to silence `SharedArrayBuffer` console warnings in E2E tests.

Updating Puppeteer is important because it contains the latest browser version that we want to test against and be compatible with.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Upgrade to puppeteer 9.1.1

## Relevant Technical Choices

Requires `npm install --force` and installing of `"@axe-core/puppeteer": "^4.1.2-alpha.118",`

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6876

### CODE REVIEW

**Comment by spacedmonkey** [comment]
I believe the issue is `@axe-core/puppeteer` see. 

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: puppeteer@9.1.1
npm ERR! node_modules/puppeteer
npm ERR!   dev puppeteer@"^9.1.0" from the root project
npm ERR!   peer puppeteer@">=1.19.0" from @wordpress/e2e-test-utils@5.1.1
npm ERR!   node_modules/@wordpress/e2e-test-utils
npm ERR!     @wordpress/e2e-test-utils@"^5.1.1" from e2e-test-utils@undefined
npm ERR!     packages/e2e-test-utils
npm ERR!     dev @wordpress/e2e-test-utils@"^5.1.1" from e2e-tests@undefined
npm ERR!     packages/e2e-tests
npm ERR!   2 more (@wordpress/jest-puppeteer-axe, @percy/puppeteer)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer puppeteer@">=1.10.0 < 6" from @axe-core/puppeteer@4.1.1
npm ERR! node_modules/@axe-core/puppeteer
npm ERR!   @axe-core/puppeteer@"^4.0.0" from @wordpress/jest-puppeteer-axe@3.0.2
npm ERR!   node_modules/@wordpress/jest-puppeteer-axe
npm ERR!     dev @wordpress/jest-puppeteer-axe@"^3.0.2" from e2e-tests@undefined
npm ERR!     packages/e2e-tests
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/jonny/.npm/eresolve-report.txt for a full report.
```
[@axe-core/puppeteer](https://github.com/dequelabs/axe-core-npm/tree/develop/packages/puppeteer) only added support for puppeteer 9.0.0 19 hours ago in https://github.com/dequelabs/axe-core-npm/pull/242. 


@swissspidy ^

**Comment by swissspidy** [comment]
Installing `@axe-core/puppeteer@next` (currently pointing at `4.1.2-alpha.118`) should resolve that, it was published 3 ours ago.

File: package-lock.json
Comment (swissspidy) [inline package-lock.json]: IDK why this gets removed from time to time, but undoing this here should fix some of the errors here.

File: package-lock.json
Comment (swissspidy) [inline package-lock.json]: Same here.

IDK why this gets removed from time to time, but undoing this change should fix some of the errors here.

**Comment by spacedmonkey** [comment]
This PR is blocked by https://github.com/google/web-stories-wp/pull/7289.

**Comment by swissspidy** [comment]
Publishing flow tests seem to be still flakey, even after #7289 being merged

**Comment by spacedmonkey** [comment]
@swissspidy It is now passing. I ran it twice and it works.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**
